packages/navigator/src/browser/navigator-widget.ts

```diff
@@ -75,6 +77,20 @@ export class FileNavigatorWidget extends FileTreeWidget {
                 this.update();
             }
         });
+        this.enableMainPanelDnd();
+    }
+
+    protected enableMainPanelDnd(): void {
+        const mainPanelNode = this.shell.mainPanel.node;
+        this.addEventListener(mainPanelNode, 'drop', e => {
+            const treeNode = this.getTreeNodeFromData(e.dataTransfer);
+            if (treeNode) {
+                const { id } = treeNode;
+                this.commandService.executeCommand(WorkspaceCommands.FILE_OPEN.id, new URI(id));
+            }
+        });
+        this.addEventListener(mainPanelNode, 'dragover', e => e.preventDefault());
```

Comment:
Should not we do it only if `e.dataTransfer` contains `tree-data`? The same for `dragenter`.

Reply:
It does not. `dragenter`, `dragover`, and `dragleave` does not contain the `tree-data`, only `drop` has the rights to check the `dataTransfer`.

Reply:
ok